Lib/bz2.py

```diff
@@ -94,6 +94,11 @@ def __init__(self, filename, mode="r", *, compresslevel=9):
         else:
             self._pos = 0
 
+        if not isinstance(filename, (str, bytes)):
+            self.name = ''
```

Comment:
Could this be some recognizable sentinel (`<stream>` or something better) rather than empty string?

Reply:
This seemed the easiest way of doing it without complicating the logic and the tests. Also, I couldn't find a uniform way of dealing with a lack of "name" for `BytesIO()` objects in other libraries, so I did not want to introduce anything "new and specific".
Do you have an example (or an idea) of a sentinel use that would fit here? In that case, we could do a similar change for `gzip`, `bz2`, `lzma`, ... libraries to have a bit more standardized way of dealing with this situation when we have `BytesIO()` objects.